File: Khelmianov-Artem/Term2/Block4/Bash/src/main/kotlin/commands/ExitCommand.kt
package commands

import bash.Environment
import utils.BashExitException
import java.io.InputStream
import java.io.OutputStream

class ExitCommand(args: List<String>) : ACommand("exit", args) {
    override fun run(input: InputStream, output: OutputStream, error: OutputStream, env: Environment): Int {
        output.write("Exiting...".toByteArray())
        val exitCode = args.getOrNull(0)?.toIntOrNull() ?: env.exitCode
        throw BashExitException(exitCode)
    }
}

File: VladislavShalnev/Term 2/Block 4/Task 8 (Bash)/Bash/Commands/ExitCommand.cs
using Bash.Exceptions;

namespace Bash.Commands;

public class ExitCommand : ICommand
{
	public string Trigger => "exit";
	
	private Action<int> _exit;

	public ExitCommand(Action<int> exit) =>
		_exit = exit;
	
	public string[] Run(string[] args)
	{
		_exit(args.Length != 0 ? int.Parse(args[0]) : 0);

		throw new ExitException();
	}
}